FinSearchUnified/Subscriber/Frontend.php

```diff
@@ -56,20 +56,18 @@ public function onFrontendPreDispatch(Enlight_Event_EventArgs $args)
         /** @var Request $request */
         $request = $subject->Request();
 
-        $controllerName = strtolower($request->getControllerName());
-
-        if ($this->isSearchPage($request)) {
-            Shopware()->Session()->offsetSet('isSearchPage', true);
-            Shopware()->Session()->offsetSet('isCategoryPage', false);
-        } elseif ($this->isCategoryPage($request)) {
-            Shopware()->Session()->offsetSet('isCategoryPage', true);
-            Shopware()->Session()->offsetSet('isSearchPage', false);
-        } elseif ($this->isManufacturerPage($request) || !(in_array($controllerName, self::WHITE_LIST, true))
-        ) {
-            Shopware()->Session()->offsetSet('isCategoryPage', false);
-            Shopware()->Session()->offsetSet('isSearchPage', false);
+        if ($this->isLegacySearch($request)) {
+            $params = $request->getQuery();
```

Comment:
You definitely need to add more whitespace, otherwise you will end up with a giant blob of code. In comparison:
```
            $params = $request->getQuery();

            unset($params['module']);
            unset($params['controller']);
            unset($params['action']);

            $url = '/search?' . http_build_query($params, null, '&', PHP_QUERY_RFC3986);

            // Perform a redirect to the actual search path and tell the client that it moved permanently.
            // The legacy parameters (relevant for search) will be mapped automatically by the corresponding
            // event handler.
            $subject->redirect($url, ['code' => 301]);
```
This also exactly how the code was provided in the story.